### ISSUE TITLE
ConvexGeometry fragility

### ISSUE THREAD

**Issue by kintel** [mrdoob/three.js#8416]
##### Description of the problem

ConvexGeometry is very fragile. It's in particular not very fond of multiple vertices aligned in almost the same plane. (follow-up from #1841).

See [live demo on CodePen](http://codepen.io/kintel/pen/ZWeWzz). Note: The test includes @marklundin's QuickHull implementation, which also fails on this geometry.

The gist of it:

```
const points = [
  new THREE.Vector3( 50, 50, -40),
  new THREE.Vector3( 50, 50, -50 ),
  new THREE.Vector3( -50, 50, -50 ),
  new THREE.Vector3( -50, 50, -40),
  new THREE.Vector3( 50, -50, -40),
  new THREE.Vector3( 50, -50, -50 ),
  new THREE.Vector3( -50, -50, -50 ),
  new THREE.Vector3( -50, -50, -40),
  new THREE.Vector3( 50, 50, 50 ),
  new THREE.Vector3( 50, 50, 40 ),
  new THREE.Vector3( -50, 50, 40 ),
  new THREE.Vector3( -50, 50, 50 ),
  new THREE.Vector3( 50, -50, 50 ),
  new THREE.Vector3( 50, -50, 40 ),
  new THREE.Vector3( -50, -50, 40 ),
  new THREE.Vector3( -50, -50, 50 ),
];

const hullgeom = new THREE.ConvexGeometry(points);
const material = new THREE.MeshPhongMaterial({ color: 0x66aaff });
const mesh = new THREE.Mesh(hullgeom, material);
```
##### Three.js version
- [x] Dev
- [x] r75
##### Browser
- [x] All of them
##### OS
- [x] All of them

**Comment by WestLangley** [comment]
That is why `ConvexGeometry` is part of the examples, I expect. If you'd like to try to improve it, that would be great.

Do you have a use case in mind -- such as collision detection?

BTW, I think `FlatShading` is  most appropriate for convex geometry.

**Comment by kintel** [comment]
Improving it is on my list :) Just just wanted to have this easily trackable and visible here.
I agree that ```FlatShading``` is more appropriate, at least until we have a good way of generating/detecting normals with crease angles.

My use-case is actually creating more complex geometry from primitives for display of OpenSCAD designs (see e.g. https://en.wikibooks.org/wiki/OpenSCAD_User_Manual/Transformations#hull). Using three.js for geometry processing is a bit of a stretch, but since I'm already using three.js for visualization, it's good for prototyping. ..and I would need the visualization anyway.

**Comment by WestLangley** [comment]
@kintel Also be aware that `Geometry.mergeVertices()`, which you are using in your demo, has a hardwired precision.

**Comment by marklundin** [comment]
Yep, coplanar sets are something that need to be checked. Should be easy to
check in qhull implementation

**Comment by ipbebnev** [comment]
How to add texture to one edge of convex geometry mesh? Can somebody write code example? 
For base we can get that: http://jsfiddle.net/cwoodall22/bk3qx8ys/

**Comment by WestLangley** [comment]
Fixed in #10987
